File: ch05/33/PrintMinNumber.cpp
class Solution {
public:
    string PrintMinNumber(vector<int> numbers) {
        string re = "";
        int len = numbers.size();
        if (0 == len)
            return re;

        sort(numbers.begin(), numbers.end(), comp);
        for (int i = 0; i < numbers.size() ; i ++) 
            re += to_string(numbers[i]); //to_string() 可以将int 转化为string
        
        return re;
    }
    
    static bool comp(int a, int b) {
        string str1 = "";
        string str2 = "";
        str1 += to_string(a);
        str1 += to_string(b);
        
        str2 += to_string(b);
        str2 += to_string(a);
        return str1 < str2;
    } 
};

File: ch05/33/PrintMinNumber.py
# -*- coding:utf-8 -*-
class Solution:
    def PrintMinNumber(self, numbers):
        # write code here
        re = ""
        if len(numbers) == 0:
            return re

        sorted_numbers = sorted(numbers, self.comp)
        for ele in sorted_numbers:
            re += str(ele)
        return re

    def comp(self, a, b):
        str1 = str(a) + str(b)
        str2 = str(b) + str(a)
        return cmp(str1, str2)


if __name__ == "__main__":
    input = [3, 32, 321]
    s = Solution()
    print s.PrintMinNumber(input)
    